test/functional/basic/element-e2e-specs.js

```diff
@@ -176,171 +174,132 @@ describe('XCUITestDriver - element(s)', function () {
+          await driver.elementSendKeys(el.elementId, '0123456789\uE006');
 
-          let text = await el.text();
+          let text = await el.getText();
           text.should.eql('0123456789');
         });
       });
 
       describe('clear', function () {
         it('should clear a text field', async function () {
           let text1 = '0123456789abcdefghijklmnopqrstuvwxyz';
-          let el = await driver.elementByClassName('XCUIElementTypeTextField');
-          await el.type(text1);
+          let el = await driver.$('XCUIElementTypeTextField');
+          await el.setValue(text1);
 
-          let text = await el.text();
+          let text = await el.getText();
           text.should.eql(text1);
 
-          await el.clear();
+          await el.clearValue();
 
-          text = await el.text();
+          text = await el.getText();
           text.should.eql(phText);
         });
         it('should be able to clear two text fields', async function () {
-          let els = await driver.elementsByClassName('XCUIElementTypeTextField');
-          await els[0].type(text1);
+          let els = await driver.$$('XCUIElementTypeTextField');
+          await els[0].setValue(text1);
 
-          let text = await els[0].text();
+          let text = await els[0].getText();
           text.should.eql(text1);
 
           await driver.hideKeyboard();
 
-          await els[1].type(text2);
+          await els[1].setValue(text2);
 
-          text = await els[1].text();
+          text = await els[1].getText();
           text.should.eql(text2);
 
-          await els[0].clear();
+          await els[0].clearValue();
 
-          text = await els[0].text();
+          text = await els[0].getText();
           text.should.eql(phText);
 
           await driver.hideKeyboard();
 
-          await els[1].clear();
+          await els[1].clearValue();
 
-          text = await els[1].text();
+          text = await els[1].getText();
           text.should.eql(phText);
         });
         it('should clear a secure text field', async function () {
-          let el = await driver.elementByClassName('XCUIElementTypeSecureTextField');
-          await el.type(text1);
+          let el = await driver.$('XCUIElementTypeSecureTextField');
+          await el.setValue(text1);
 
-          let text = await el.text();
+          let text = await el.getText();
           text.should.eql(secureText);
 
-          await el.clear();
-          text = await el.text();
+          await el.clearValue();
+          text = await el.getText();
           text.should.eql(phText);
         });
       });
-      describe('keys', function () {
-        it('should be able to send text to the active element', async function () {
-          let el = await driver.elementByClassName('XCUIElementTypeTextField');
-          // make sure the keyboard is up
-          await el.click();
-
-          await driver.keys('this is a test');
```

Comment:
Sending keys to an active element can do with `actions` like... https://github.com/appium/ruby_lib_core/blob/1632637fd872c0b80dfb97b8514ada6a7164eebf/test/unit/android/webdriver/w3c/actions_test.rb#L77-L113

Maybe https://webdriver.io/docs/api/webdriver/#performactions is in webdriverio.

Reply:
thank you for the info. I added a test case to sending a text to active element. BTW, can we send a unicode key(?) like a backspace key? As far as I checked docs, it looks possible (Ref: https://appium.io/docs/en/commands/interactions/actions/, https://github.com/jlipps/simple-wd-spec#perform-actions) However, it doesn't work as I expected.

fyi, I tried as the following piece of code 
```
it('should key a backspace', async function () {
  let el = await driver.$('XCUIElementTypeTextField');
  // make sure the keyboard is up
  await el.click();

  const actions = [{
    type: 'key',
    id: 'keyboard',
    actions: [
      {type: 'keyDown', value: '0'},
      {type: 'keyUp', value: '0'},
      {type: 'keyDown', value: '1'},
      {type: 'keyUp', value: '1'},
      {type: 'keyDown', value: '\uE003'},
      {type: 'keyUp', value: '\uE003'}
    ]
  }];
  await driver.performActions(actions);

  let text = await el.getText();
  text.should.eql('0');
})
```

However, it doesn't work as 
```
2022-11-03T22:54:52.944Z INFO webdriver: COMMAND elementClick("16070000-0000-0000-CA02-010000000000")
2022-11-03T22:54:52.944Z INFO webdriver: [POST] http://127.0.0.1:4567/session/c437d218-99b5-4c4c-a2c7-f874bfaa94bf/element/16070000-0000-0000-CA02-010000000000/click
2022-11-03T22:54:53.689Z INFO webdriver: COMMAND performActions(<object>)
2022-11-03T22:54:53.692Z INFO webdriver: [POST] http://127.0.0.1:4567/session/c437d218-99b5-4c4c-a2c7-f874bfaa94bf/actions
2022-11-03T22:54:53.692Z INFO webdriver: DATA { actions: [ { type: 'key', id: 'keyboard', actions: [Array] } ] }
2022-11-03T22:54:59.060Z INFO webdriver: COMMAND getElementText("16070000-0000-0000-CA02-010000000000")
2022-11-03T22:54:59.061Z INFO webdriver: [GET] http://127.0.0.1:4567/session/c437d218-99b5-4c4c-a2c7-f874bfaa94bf/element/16070000-0000-0000-CA02-010000000000/text
2022-11-03T22:54:59.148Z INFO webdriver: RESULT Placeholder text
          1) should key a backspace
2022-11-03T22:54:59.164Z INFO webdriver: COMMAND back()
2022-11-03T22:54:59.164Z INFO webdriver: [POST] http://127.0.0.1:4567/session/c437d218-99b5-4c4c-a2c7-f874bfaa94bf/back
2022-11-03T22:55:00.098Z INFO webdriver: COMMAND deleteSession()
2022-11-03T22:55:00.100Z INFO webdriver: [DELETE] http://127.0.0.1:4567/session/c437d218-99b5-4c4c-a2c7-f874bfaa94bf


  0 passing (28s)
  1 failing

  1) XCUITestDriver - elements -
       interactions
         text fields
           key
             should key a backspace:

      AssertionError: expected 'Placeholder text' to deeply equal '0'
      + expected - actual

      -Placeholder text
      +0
      
      at Context.<anonymous> (test/functional/basic/element-e2e-specs.js:345:23)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Reply:
Probably we haven't tested that case, or it can with `\n`
Could you create an issue?